diff/parse.go

```diff
@@ -579,6 +593,16 @@ func linePrefix(c byte) bool {
 	return false
 }
 
+// nextLineHasPrefix peeks into the given reader to check whether the next
+// bytes match the given prefix.
+func nextLineHasPrefix(reader *bufio.Reader, prefix []byte) (bool, error) {
+	next, err := reader.Peek(len(prefix))
```

Comment:
is it possible that calling Peek affects the bytes previously returned? eg in some cases I know when using a reader it returns a slice directly from the buffer. So I'm worried about a call to peek mutating that buffer, which in turn mutates a slice that is used above. I'm not familiar enough with this code to know this offhand, so just calling it out since you mutated and should be able to say if this is safe.

Reply:
I'm 99% sure this is safe. The code is not exactly straightforward, but as far as I can tell, the reader is only used to pass it to `readLine` and the resulting line slice is then modified cut in place. Having read through `Peek` and what it does, I can't see how it affects the previous read lines, _especially_ since we're on the new line.